sdk/src/clearingHouse.ts

```diff
@@ -611,32 +801,88 @@ export class ClearingHouse {
 	 */
 	public async initializeUserAccountAndDepositCollateral(
 		amount: BN,
-		userTokenAccount: PublicKey,
-		bankIndex = new BN(0),
+		authority: PublicKey,
+		bank: BankConfig,
```

Comment:
I'd prefer to keep this as bankIndex. Some of my reasons:
1) we use market index every where else so its more consistent
2) eventually i'd like to have a way for users to query for bank/markets via a web api so that they don't have to load the new sdk to get new markets. I think passing in the bank/market index to these functions is less opinionated and thus would make a transition to this easier
3) i wont need to go back and change all the tests to pass in a bank config. for anchor tests, sometimes we spawn multiple banks and we dont have a BankConfig for them in the config file